subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/windowsStartScript.txt

```diff
@@ -37,6 +37,12 @@ for %%i in ("%APP_HOME%") do set APP_HOME=%%~fi
 set DEFAULT_JVM_OPTS=${defaultJvmOpts}
 
 @rem Find java.exe
+if exist %APP_HOME%/jre/bin/java.exe (
```

Comment:
This looks like a "magic convention" for me. I don't see this convention in any other tools, can you please give me some examples of other tools using this convention?

I wouldn't include this in core logic, but would like write a plugin to customize it.

Reply:
Just like the jre that comes with the directory of the AndroidStudio program:

![YeMrBF.png](https://s1.ax1x.com/2020/05/07/YeMrBF.png)

Reply:
> This looks like a "magic convention" for me. I don't see this convention in any other tools, can you please give me some examples of other tools using this convention?
> 
> I wouldn't include this in core logic, but would like write a plugin to customize it.

Just like the jre that comes with the directory of the AndroidStudio program:
![YeMrBF.png](https://s1.ax1x.com/2020/05/07/YeMrBF.png)